packages/cli-kit/src/node/base-command.test.ts

```diff
@@ -0,0 +1,304 @@
+import Command from './base-command.js'
+import {Presets, presetsFilename} from '../public/node/presets.js'
+import {globalFlags} from '../cli.js'
+import {mkdir, mkTmpDir, rmdir, write as writeFile} from '../file.js'
+import {mockAndCaptureOutput} from '../testing/output.js'
+import {encode as encodeTOML} from '../toml.js'
+import {join as pathJoin, resolve as resolvePath} from '../path.js'
+import {afterEach, beforeEach, describe, expect, test} from 'vitest'
+import {Flags} from '@oclif/core'
+
+let testResult: {[flag: string]: unknown} = {}
+let testError: Error | undefined
+let disableFindUpPresets = true
+
+class MockCommand extends Command {
+  /* eslint-disable rulesdir/command-flags-with-env */
+  static flags = {
+    ...globalFlags,
+    path: Flags.string({
+      parse: (input, _) => Promise.resolve(resolvePath(input)),
+      default: '.',
+    }),
+    someString: Flags.string({}),
+    someInteger: Flags.integer({}),
+    someBoolean: Flags.boolean({
+      allowNo: true,
+    }),
+    someExclusiveBoolean: Flags.boolean({
+      exclusive: ['someBoolean'],
+    }),
+    someMultipleString: Flags.string({multiple: true}),
+    someStringWithDefault: Flags.string({
+      default: 'default stringy',
+    }),
+  }
+  /* eslint-enable rulesdir/command-flags-with-env */
+
+  async run(): Promise<void> {
+    const {flags} = await this.parse(MockCommand)
+    testResult = flags
+  }
+
+  async catch(error: Error): Promise<void> {
+    testError = error
+  }
+
+  findUpForPresets() {
+    if (disableFindUpPresets) return false
+    return super.findUpForPresets()
+  }
+}
+
+const validPreset = {
+  someString: 'stringy',
+  someBoolean: true,
+}
+
+const validPresetWithIrrelevantFlag = {
+  ...validPreset,
+  irrelevantString: 'stringy',
+}
+
+const presetWithIncorrectType = {
+  someInteger: 'stringy',
+}
+
+const presetWithExclusiveArguments = {
+  someBoolean: true,
+  someExclusiveBoolean: false,
+}
+
+const presetWithNegativeBoolean = {
+  someBoolean: false,
+}
+
+const presetWithMultiples = {
+  someMultipleString: ['multiple', 'stringies'],
+}
+
+const presetMatchingDefault = {
+  someStringWithDefault: 'default stringy',
+}
+
+const presetWithDefaultOverride = {
+  someStringWithDefault: 'non-default stringy',
+}
+
+const allPresets: Presets = {
+  validPreset,
+  validPresetWithIrrelevantFlag,
+  presetWithIncorrectType,
+  presetWithExclusiveArguments,
+  presetWithNegativeBoolean,
+  presetWithMultiples,
+  presetMatchingDefault,
+  presetWithDefaultOverride,
+}
+
+describe('applying presets', async () => {
+  let tmpDir: string
+
+  beforeEach(async () => {
+    tmpDir = await mkTmpDir()
+    await writeFile(pathJoin(tmpDir, presetsFilename), encodeTOML(allPresets as any))
+    disableFindUpPresets = false
+  })
+
+  afterEach(async () => {
+    if (tmpDir) {
+      await rmdir(tmpDir)
```

Comment:
We do it here https://github.com/Shopify/cli/blob/45e817b9cd1d482fb2912a58ab27c652c0f05810/packages/app/src/cli/models/app/loader.test.ts#L15-L23 so I assumed it's fine.

The goal is to give each test a fresh directory to work with, as the directory may be modified during the test.

It's true that the tmpDir wouldn't be removed, it's a price I'm willing to pay as it's a tmpDir anyway so it's supposed to be prone to being removed by the OS, right?

Reply:
I see 🤔 
That might be a legacy use (is the only one in the code), if you search for `inTemporaryDirectory` you'd see is the common way to do it and I think it's safer, because the tmp folder won't be cleaned up by the OS afaik